neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jResultSet.java

```diff
@@ -91,8 +92,14 @@ private BoltNeo4jResultSet(Statement statement, StatementResult iterator, int...
 			this.flattenResultSet();
 			this.flattened = true;
 		} else if (this.iterator != null) {
-			//Keys are exactly the ones returned from the iterator
-			this.keys = this.iterator.keys();
+			// Keys are exactly the ones returned from the iterator
+			// This may raise an exception if the statement is invalid
+			try {
+				this.keys = this.iterator.keys();
+			}
+			catch (ClientException ce) {
+				throw new SQLException(ce.getMessage(), ce);
+			}
```

Comment:
I would move the try-catch at line 89 wrapping the whole `if/else/if` block. WDYT?

Reply:
that's probably safer indeed :)
let me try it